x/bsonx/bsoncore/document_test.go

```diff
@@ -413,127 +412,67 @@ func TestDocument(t *testing.T) {
-		), ""},
-
-		{"n<0, document with mixed types", -1, BuildDocument(nil,
-			AppendStringElement(nil, "key", str128),
-			AppendInt32Element(nil, "number", 123),
-		), ""},
-
-		{"n<0, deeply nested document", -1, BuildDocument(nil,
-			AppendDocumentElement(nil, "a", BuildDocument(nil,
-				AppendDocumentElement(nil, "b", BuildDocument(nil,
-					AppendStringElement(nil, "c", str128),
-				)),
-			)),
-		), ""},
-
-		{"n<0, complex value", -1, BuildDocument(nil,
-			AppendDocumentElement(nil, "key", BuildDocument(nil,
-				AppendStringElement(nil, "nestedKey", str128),
-			)),
-		), ""},
-
-		// n > 0 cases
-		{"n>0, document LT n", 3, BuildDocument(nil,
-			AppendStringElement(nil, "key", "value"),
-		), `{"k`},
-
-		{"n>0, document GT n", 25, BuildDocument(nil,
-			AppendStringElement(nil, "key", "value"),
-		), `{"key": "value"}`},
-
-		{"n>0, document EQ n", 16, BuildDocument(nil,
-			AppendStringElement(nil, "key", "value"),
-		), `{"key": "value"}`},
-
-		{"n>0, document with nested documents", 15, BuildDocument(nil,
-			AppendDocumentElement(nil, "key", BuildDocument(nil,
-				AppendStringElement(nil, "nestedKey", str128),
-			)),
-		), `{"key": {"neste`},
-
-		{"n>0, document with mixed types", 11, BuildDocument(nil,
-			AppendStringElement(nil, "key", str128),
-			AppendInt32Element(nil, "number", 123),
-		), `{"key": "ab`},
-
-		{"n>0, deeply nested document", 17, BuildDocument(nil,
-			AppendDocumentElement(nil, "a", BuildDocument(nil,
-				AppendDocumentElement(nil, "b", BuildDocument(nil,
-					AppendStringElement(nil, "c", str128),
+		{
+			description: "empty document",
+			doc:         BuildDocument(nil),
+			want:        `{}`,
+		},
+		{
+			description: "document with 1 field",
+			doc: BuildDocument(nil,
+				AppendInt32Element(nil, "number", 123),
+			),
+			want: `{"number": {"$numberInt":"123"}}`,
+		},
+		{
+			description: "nested documents",
+			doc: BuildDocument(nil,
+				AppendDocumentElement(nil, "key", BuildDocument(nil,
+					AppendStringElement(nil, "nestedKey", "abc"),
 				)),
-			)),
-		), `{"a": {"b": {"c":`},
-
-		{"n>0, empty document", 10, Document{}, ""},
+			),
+			want: `{"key": {"nestedKey": "abc"}}`,
+		},
+
+		{
+			description: "document with mixed types",
+			doc: BuildDocument(nil,
+				AppendStringElement(nil, "key", "abc"),
+				AppendInt32Element(nil, "number", 123),
+				AppendBooleanElement(nil, "flag", true),
+			),
+			want: `{"key": "abc","number": {"$numberInt":"123"},"flag": true}`,
+		},
 	}
 
 	for _, tc := range testCases {
-		t.Run(tc.description, func(t *testing.T) {
-			bs := tc.doc
-			got := bs.StringN(tc.n)
+		t.Run(fmt.Sprintf("String %s", tc.description), func(t *testing.T) {
+			got := tc.doc.String()
 			assert.Equal(t, tc.want, got)
-			if tc.n >= 0 {
-				assert.LessOrEqual(t, len(got), tc.n)
-			}
 		})
 	}
+
+	for _, tc := range testCases {
+		for n := -1; n <= len(tc.want)+1; n++ {
+			t.Run(fmt.Sprintf("StringN %s n==%d", tc.description, n), func(t *testing.T) {
```

Comment:
Optional: Consider factoring these into two test functions `TestDocument_String` and `TestDocument_StringN` and moving the test cases to a package-level variable (e.g. `var documentStringTestCases = ...`).